### PR TITLE
Fix historical network supply

### LINKED ISSUE #7441: Incorrect historical network supply
### Description

After balance deduplication, we did not update the network supply API when a timestamp is supplied.

### Steps to reproduce

curl -s https://mainnet-public.mirrornode.hedera.com/api/v1/network/supply

```json
{
  "released_supply": "3361139192880807845", 
  "timestamp": "1703198615.991171003",
  "total_supply": "5000000000000000000"
}
```

curl -s https://mainnet-public.mirrornode.hedera.com/api/v1/network/supply?timestamp=1703198615.991171003

```json
{
  "released_supply": "4949999986070886882",
  "timestamp": "1703198027.998830003",
  "total_supply": "5000000000000000000"
}
```

### Additional context

_No response_

### Hedera network

mainnet

### Version

v0.94.1

### Operating system

None

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:

* Update the `/api/v1/network/supply?timestamp=` query for deduplicated balances
* Update the `/api/v1/network/supply` to drop the join on `record_file` in favor of using max `balance_timestamp`
* Update the tables in `/docs/rest/README.md`
* Update `unreleasedSupplyAccounts()` to optimize `account_balance >= X and account_balance <= X` to `account_balance = X`.

**Related issue(s)**:

Fixes #7441

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by edwin-greene — APPROVED** [review]
Looks good

**Review by mgoelswirlds — APPROVED** [review]
LGTM

File: hedera-mirror-rest/__tests__/specs/network/supply/no-accounts.json [hunk 1]
```diff
@@ -1,27 +1,17 @@
...
-        "index": 16,
-        "name": "2022-04-27T12_09_24.499938763Z.rcd",
-        "prev_hash": "abc"
-      }
-    ]
+    "accounts": []
   },
   "url": "/api/v1/network/supply",
-  "responseStatus": 200,
+  "responseStatus": 404,
```
Comment (steven-sheehy) [inline hedera-mirror-rest/__tests__/specs/network/supply/no-accounts.json:7]: We removed the join on record_file so query now returns no results and thus 404. It didn't make sense that it returned results for no accounts before.

**Review by jascks — APPROVED** [review]
LGTM

**Pull request merged**